From f662947f7bd03b10baf2899f980b3fb86d6f6639 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Thu, 15 Dec 2011 20:46:52 +0200
Subject: [PATCH] Resolved issue with local variable in singleton

bcc needs a local object (stack) to have its destructor public
even when created from within scope. (bcc32 bug)
---
 src/BRepMesh/BRepMesh_DiscretFactory.cxx |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/src/BRepMesh/BRepMesh_DiscretFactory.cxx b/src/BRepMesh/BRepMesh_DiscretFactory.cxx
index 69f3402..6d4ce30 100644
--- a/src/BRepMesh/BRepMesh_DiscretFactory.cxx
+++ b/src/BRepMesh/BRepMesh_DiscretFactory.cxx
@@ -74,9 +74,14 @@ void BRepMesh_DiscretFactory::Clear()
 //=======================================================================
 BRepMesh_DiscretFactory& BRepMesh_DiscretFactory::Get()
 {
+#ifdef __BORLANDC__
+  static BRepMesh_DiscretFactory* sFactory = new BRepMesh_DiscretFactory;
+  return *sFactory;
+#else
   //! global factory instance
   static BRepMesh_DiscretFactory THE_GLOBAL_FACTORY;
   return THE_GLOBAL_FACTORY;
+#endif
 }
 
 //=======================================================================
-- 
1.7.9.1

